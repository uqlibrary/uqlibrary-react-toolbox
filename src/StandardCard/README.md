# StandardCard

component for a standard card display with help button

## Props

- `title: PropTypes.string.isRequired`
- `help: PropTypes.object`, eg {title: 'About', text: 'help text..', buttonLabel:'Close'}

## Usage

**./App.js**
```jsx
import React from 'react';
...

import {StandardCard} from 'uqlibrary-react-toolbox';
        
const App = () => (
  <div>
    <StandardCard 
      title='Contact us' 
      text='contact details here...' 
      help={title: 'About', text: 'help text..', buttonLabel:'Close'} 
       />
  </div>
);

ReactDOM.render(
  <App />,
  document.getElementById('app')
);
```

