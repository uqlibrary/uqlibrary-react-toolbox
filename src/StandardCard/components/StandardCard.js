import React from 'react';
import PropTypes from 'prop-types';

import {Card, CardText, CardHeader} from 'material-ui/Card';
import {HelpIcon} from '../../index';

export default function StandardCard({title, help, className, children}) {
    return (
        <Card className={`layout-card ${className}`}>
            <CardHeader className="card-header">
                <div className="columns is-gapless is-mobile">
                    <div className="column">
                        {title && <h2 className="title is-4">{title}</h2>}
                    </div>
                    {help && help.text &&
                    <div className="column is-narrow is-helpicon">
                        <HelpIcon {...help}/>
                    </div>
                    }
                </div>
            </CardHeader>
            <CardText className="body-1">
                <div>
                    {children}
                </div>
            </CardText>
        </Card>
    );
}

StandardCard.propTypes = {
    title: PropTypes.string,
    children: PropTypes.any,
    className: PropTypes.string,
    help: PropTypes.shape({
        title: PropTypes.string,
        text: PropTypes.any,
        buttonLabel: PropTypes.string
    })
};
