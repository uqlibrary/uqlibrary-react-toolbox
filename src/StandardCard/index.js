export StandardCard from './components/StandardCard';
