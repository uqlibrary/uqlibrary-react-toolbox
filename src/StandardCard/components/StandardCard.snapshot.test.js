jest.dontMock('./StandardCard');

import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import StandardCard from './StandardCard';

function setup(title, help) {
    const props = {title, help};
    return shallow(<StandardCard {...props} />);
}

describe('Snapshot tests for StandardCard component', () => {
    it('renders StandardCard with title and no help icon', () => {
        const wrapper = setup('card title');
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

    it('renders StandardCard with title and help button', () => {
        const wrapper = setup('card title', {title: 'help', text: 'help text', buttonLabel: 'OK'});
        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});