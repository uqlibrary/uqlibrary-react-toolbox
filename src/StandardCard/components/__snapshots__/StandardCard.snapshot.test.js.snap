// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests for StandardCard component renders StandardCard with title and help button 1`] = `
<Card
  className=" standard-card"
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    avatar={null}
    className="card-header"
  >
    <div
      className="columns is-gapless is-mobile"
    >
      <div
        className="column"
      >
        <h2
          className="cardTitle"
        >
          card title
        </h2>
      </div>
      <div
        className="column is-narrow is-helpicon"
      >
        <Connect(HelpIcon)
          buttonLabel="OK"
          text="help text"
          title="help"
        />
      </div>
    </div>
  </CardHeader>
  <CardText
    className="body-1"
  >
    <div />
  </CardText>
</Card>
`;

exports[`Snapshot tests for StandardCard component renders StandardCard with title and no help icon 1`] = `
<Card
  className=" standard-card"
  expandable={false}
  expanded={null}
  initiallyExpanded={false}
>
  <CardHeader
    avatar={null}
    className="card-header"
  >
    <div
      className="columns is-gapless is-mobile"
    >
      <div
        className="column"
      >
        <h2
          className="cardTitle"
        >
          card title
        </h2>
      </div>
    </div>
  </CardHeader>
  <CardText
    className="body-1"
  >
    <div />
  </CardText>
</Card>
`;
